lib/wibox/drawable.lua

```diff
@@ -128,12 +130,25 @@ local function do_redraw(self)
 
     cr:set_source(self.background_color)
     cr:paint()
+
+    -- Paint the background image
+    if self.background_image then
+        if type(self.background_image) == "function" then
+            --TODO if opaque, cache this
```

Comment:
Why? I don't think caching this will be significantly faster and if every drawable caches this, the memory usage might be significant.

I don't agree with this "TODO".